### ISSUE TITLE
Unable to create Meta Data for dialects in file manager

### ISSUE THREAD

**Issue by fritzmg** [contao/core#7667]
#7112 created a side effect: you cannot define meta data for dialects (subtags) anymore. 

### Reproduction

Create a simple website structure with two website roots:

* one with language __de__
* one with language __de-AT__ for example

You are only able to create meta data of files in the file manager for __de (German)__, not for __de-AT (German (Austria))__ as well.

### Cause

[MetaWizard::generate()](https://github.com/contao/core/blob/master/system/modules/core/widgets/MetaWizard.php#L67) does the following:

It gets the supported languages by calling [System::getLanguages()](https://github.com/contao/core/blob/master/system/modules/core/library/Contao/System.php#L406), this will return an array with entries like these:
```php
array(
    ...
    "de" => "German"
    "de_DE" => "German (Germany)"
    "de_AT" => "German (Austria)"
    "de_CH" => "German (Switzerland)"
    ...
);
```
Further on, all the website root pages are fetched and their configured languages. In our case the languages of the website roots will be:
```php
array(
    'de',
    'de-AT'
);
```
The function then determines the allowed languages for the meta data by intersecting the keys of both arrays (the latter one being flipped beforehand).

Since in the original languages array from the system, the German Austrian language tag and subtag is __de_AT__, but the language setting of the root page is __de-AT__ (you are not allowed to actually enter `de_AT` instead of `de-AT`), the resulting intersected array will only contain
```php
array(
    'de'
);
```

### Proposed fix

Well, to be honest, I don't know what the best fix here would be, may be a preg_match on all elements, instead of `array_intersect_key()`?

As a side note, #7112 also causes Problems for the i18ln10 extension (see [here](https://community.contao.org/de/showthread.php?48943-i18nl10n-Neuer-Entwickler-gefunden&p=360873&viewfull=1#post360873)).

**Comment by thomaspahl** [comment]
According to [W3C]([http://www.w3.org/International/articles/language-tags/), RFC 3066, and the [IANA langauge tag registry](http://www.w3.org/International/articles/language-tags/) the separator should be `-` and not `_` at least in the web -- whereas the POSIX locale uses `_`. So the conflict is beyond Contao.

A possible solution would be to use `array_intersect_ukey()` with a callback function that does a tolerant compare (case independent as well).
A simple callback would be (not tested):
```
function language_compare($lang1, $lang2)
{
	if ($lang1 == $lang2)  // questionable optimization...
	{
		return 0;
	}
	return strcasecmp(str_replace('_', '-', $lang1), str_replace('_', '-', $lang2));
}
```

**Comment by fritzmg** [comment]
Sounds good. Maybe it would even make sense to extend the [System::getLanguages()](https://github.com/contao/core/blob/master/system/modules/core/library/Contao/System.php#L406) function like so:
```php
public static function getLanguages($blnInstalledOnly=false, $blnPosix=true)
```
If `$blnPosix` is false, the array will be converted accordingly before it is returned.

**Comment by leofeyer** [comment]
Fixed in cd0d256c69a49151c4a3d2ab0a8de503a6dd8294.
